src/prototype_creep_resources.js

```diff
@@ -1,9 +1,20 @@
 'use strict';
 
 Creep.prototype.harvesterBeforeStorage = function() {
-  var methods = [
-    Creep.getEnergy
-  ];
+  let methods = [];
+
+  if (this.memory.hasEnergy === undefined) {
+    this.memory.hasEnergy = (this.carry.energy == this.carryCapacity);
+  } else if (this.memory.hasEnergy && this.carry.energy === 0) {
+    this.memory.hasEnergy = false;
+  } else if (!this.memory.hasEnergy &&
+      this.carry.energy == this.carryCapacity) {
+    this.memory.hasEnergy = true;
+  }
```

Comment:
I still think that this will prevent the harvester from getting energy out of the sources